pkg/proc/stack.go

```diff
@@ -252,8 +252,15 @@ func (it *stackIterator) Next() bool {
 	}
 
 	if it.opts&StacktraceSimple == 0 {
-		if it.switchStack() {
-			return true
+		switch it.bi.Arch.(type) {
```

Comment:
Add a `SwitchStack` method to Arch and make amd64switchStack and arm64switchStack methods of ARM64 and AMD64.

Reply:
modified.